rand_core/src/lib.rs

```diff
@@ -185,22 +207,20 @@ pub trait SeedableRng<Seed>: Rng {
     fn from_seed(seed: Seed) -> Self;
 }
 
-
-/// Error type for cryptographic generators. Technically external generators
-/// such as the operating system or hardware generators could fail. A PRNG
-/// (algorithm) could also fail if it detects cycles, though most PRNGs have
-/// sufficiently long cycles that looping is not usually feasible.
-/// 
-/// TODO: how should error details be reported?
-#[derive(Debug)]
-pub struct Error;
-
-#[cfg(feature="std")]
-impl From<::std::io::Error> for Error {
-    fn from(_: ::std::io::Error) -> Error {
-        Error
-    }
+/// Error type for random number generators.
+#[derive(Copy, PartialEq, Eq, Clone, Debug)]
+pub enum RngError {
```

Comment:
I'd prefer the name `Error` personally — less repetition and matches usual naming conventions.

Reply:
I thought I was following some sort of convention after reading the [Error Handling](https://doc.rust-lang.org/book/first-edition/error-handling.html#standard-library-traits-used-for-error-handling) Chapter in TRPL again. But `Error` seems to be common, and it shorter. I will change it.

Also just noticed: It should implement the `::std::error::Error` trait.